akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala

```diff
@@ -97,6 +99,21 @@ import akka.util.JavaDurationConverters._
     }(responseTimeout.asScala, ClassTag[Res](resClass))
   }
 
+  // Scala API impl
+  def pipeToSelf[Value](future: Future[Value])(mapResult: Try[Value] ⇒ T): Unit = {
+    future.onComplete(value ⇒ self.unsafeUpcast ! AdaptMessage(value, mapResult))
+  }
+
+  // Java API impl
+  def pipeToSelf[Value](future: CompletionStage[Value], applyToResult: BiFunction[Value, Throwable, T]): Unit = {
+    future.whenComplete(new BiConsumer[Value, Throwable] {
+      def accept(value: Value, ex: Throwable): Unit = {
+        if (value != null) self ! applyToResult.apply(value, null)
+        if (ex != null) self ! applyToResult.apply(null.asInstanceOf[Value], ex)
+      }
+    })
+  }
+
```

Comment:
Not super important but, cannot context.ask now be expressed using ask plus `pipeToSelf` to avoid a little bit of duplication?